### ISSUE TITLE
Multi-cluster scenario fails unless an update is forced (1.3.x regression)

### ISSUE THREAD

**Issue by ericvn** [istio/istio#17562]
Running through the instructions multi-cluster, shared control plane example on my two clusters and the example works as described.

There is another example in the documentation that uses a BookInfo deployment on cluster 1, and the review-v3 app is stopped, but started on cluster 2. In the example, hitting the product page should result in all 3 reviews being hit and alternating between red/black/no stars being shown.

This does not work. I only see no/black starts. the v3 app on cluster 2 is never called.

I can get this to work if I do two things:
 - Update the reviews-v3.yaml to be more like the bookinfo.yaml (I took the review and ratings sections from bookinfo, removed the ratings, reviews-v1, and reviews-v2 deployments).
 - Force an update by applying the default destination rules and removing them. Will the rules were active the app failed to run.

It maybe that the second item isn't needed, and would happen at some point.

**Affected product area (please put an X in all that apply)**

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ x ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

**Expected behavior**

**Steps to reproduce the bug**

**Version (include the output of `istioctl version --remote` and `kubectl version`)**

1.3.1

**How was Istio installed?**

Installed using default (non-tiller) instructions on two IKS clusters

**Environment where bug was observed (cloud vendor, OS, etc)**

Additionally, please consider attaching a [cluster state archive](http://istio.io/help/bugs/#generating-a-cluster-state-archive) by attaching
the dump file to this issue.

**Comment by ericvn** [comment]
@howardjohn @sdake mentioned you might be interested in this.

**Comment by frankbu** [comment]
There may be more than one problem here, but one thing that is definitely wrong is that the config in step 2 of [the instructions](https://preliminary.istio.io/docs/examples/multicluster/gke/#deploy-bookinfo-example-across-clusters) is a using a different version than what's used in step 1 (bookinfo.yaml). So the doc needs to be updated.

To avoid this problem in the future, I would suggest replacing step 2 and 3 with the following:

```
kubectl apply -f samples/bookinfo/platform/kube/bookinfo.yaml -l service=ratings
kubectl apply -f samples/bookinfo/platform/kube/bookinfo.yaml -l service=reviews
kubectl apply -f samples/bookinfo/platform/kube/bookinfo.yaml -l app=reviews,version=v3
```

This way the versions on the remote cluster are guaranteed to be consistent with the local one.

**Comment by ericvn** [comment]
Thanks @frankbu . I think that would fix the first part. Still two things to investigate:

1) Why do we need to apply/delete the DR to get things to work? Is this causing a config update that should have happened normally
2) Why does applying the DRs cause the application to fail?

**Comment by ericvn** [comment]
With the DR applied, I see
```
[2019-10-02 17:56:05.255][16][warning][config] [bazel-out/k8-opt/bin/external/envoy/source/common/config/_virtual_includes/grpc_stream_lib/common/config/grpc_stream.h:87] gRPC config stream closed: 13,
[2019-10-02 17:57:40.181][16][warning][config] [bazel-out/k8-opt/bin/external/envoy/source/common/config/_virtual_includes/grpc_stream_lib/common/config/grpc_stream.h:87] gRPC config stream closed: 13,
[2019-10-02 18:02:40.215][16][warning][config] [bazel-out/k8-opt/bin/external/envoy/source/common/config/_virtual_includes/grpc_stream_lib/common/config/grpc_stream.h:87] gRPC config stream closed: 13,
[2019-10-02 18:07:40.442][16][warning][config] [bazel-out/k8-opt/bin/external/envoy/source/common/config/_virtual_includes/grpc_stream_lib/common/config/grpc_stream.h:87] gRPC config stream closed: 13,
[2019-10-02 18:12:40.870][16][warning][config] [bazel-out/k8-opt/bin/external/envoy/source/common/config/_virtual_includes/grpc_stream_lib/common/config/grpc_stream.h:87] gRPC config stream closed: 13,
[2019-10-02 18:17:41.231][16][warning][config] [bazel-out/k8-opt/bin/external/envoy/source/common/config/_virtual_includes/grpc_stream_lib/common/config/grpc_stream.h:87] gRPC config stream closed: 13,
[2019-10-02 18:25:32.275][16][warning][config] [bazel-out/k8-opt/bin/external/envoy/source/common/config/_virtual_includes/grpc_stream_lib/common/config/grpc_stream.h:87] gRPC config stream closed: 13,
[2019-10-02 18:29:15.165][16][warning][config] [bazel-out/k8-opt/bin/external/envoy/source/common/config/_virtual_includes/grpc_stream_lib/common/config/grpc_stream.h:87] gRPC config stream closed: 13,
[2019-10-02 18:34:15.202][16][warning][config] [bazel-out/k8-opt/bin/external/envoy/source/common/config/_virtual_includes/grpc_stream_lib/common/config/grpc_stream.h:87] gRPC config stream closed: 13,
``` in the logs. Related to #17383

**Comment by frankbu** [comment]
@ericvn What DR are you applying? Are you applying it in both clusters?

**Comment by ericvn** [comment]
You can only apply to the main cluster as it has the control plane.

**Comment by ericvn** [comment]
NOTE that this is a regression from 1.2.6.  On 1.2.6, I was able to get this running without needing to apply/delete the DRs. For this case I used my changed yaml. Your example changes above @frankbu did not cause the v3 pod to be started. The deployment was 0/1.

**Comment by frankbu** [comment]
@ericvn fixes in https://github.com/istio/istio/pull/17577 and https://github.com/istio/istio.io/pull/5094 should bring up the v3 pod successfully. Can you try it, and if it works approve both PRs? Thanks.

**Comment by ericvn** [comment]
@frankbu Tested changes on 1.2.6 and things came up and we had traffic to all reviews so from a doc standpoint this looks good.

We still have an issue on 1.3.x where something needs to trigger an update of the config to get traffic to flow to reviews-v3.

Also there is still an issue with applying the DRs in this multi cluster configuration causing the app to fail. This issue might warrant the creation of another issue, leaving this one covering the regression from 1.2 to 1.3 of the requirement of an update (the original issue title).

**Comment by frankbu** [comment]
Thanks @ericvn. I agree that we should open a separate issue about DRs not working in the multicluster env. I wonder if it's only a problem in the multi-network (gateways) config? Anyway, if you can open an issue describing how to setup the env to illustrate the problem you've seen, that would be great.

**Comment by ericvn** [comment]
Issue opened for failure after applying default DRs in BookInfo: https://github.com/istio/istio/issues/17591

**Comment by ericvn** [comment]
I did slack @howardjohn and decided to try and run the scenario yet again on Istio 1.3.1 against my IKS clusters. I generated some dumps after setting up Istio and BookInfo on the cluster. This was with BookInfo installed on cluster 1 with reviews-v3 removed and just reviews v3 was placed on cluster 2. I did use the newer changes in preliminary.istio for the deploy in cluster 2 (see above for issue numbers).

Cluster 1:
https://drive.google.com/open?id=1yQ9W7jgHU6myWWlhaI_jODWauRjaI7uA

Cluster 2:
https://drive.google.com/open?id=1dsjvNG9IAPzgPz_Io4TXYfFFgx9x2DdC

John also mentioned that this example does not set the cluster name helm flag.

I did exec into pilot on both clusters and tried a push to see if things changed: `curl localhost:8080/debug/adsz?push=true`. They did not.

I applied the _destination-rule-all-mtls.yaml_ to both clusters and still nothing hits the cluster #2 reviews-v3.

I deleted the DR from both clusters and still no traffic to cluster 2.

I then tried applying _destination-rule-all.yaml_ to cluster 1 which caused errors in the app earlier, and deleted the yaml and still no traffic (this caused traffic to flow earlier)...

Now I'm confused things don't eventually flow...
